pulpcore/cli/core/vulnerability_report.py

```diff
@@ -0,0 +1,73 @@
+import typing as t
+
+import click
+from pulp_glue.common.context import PulpRepositoryContext
+from pulp_glue.common.i18n import get_translation
+from pulp_glue.core.context import PulpVulnerabilityReportContext
+from pulp_glue.python.context import PulpPythonRepositoryVersionContext
+
+from pulpcore.cli.common.generic import (
+    PulpCLIContext,
+    href_option,
+    list_command,
+    pass_pulp_context,
+    pulp_group,
+    resource_option,
+    show_command,
+)
+
+translation = get_translation(__package__)
+_ = translation.gettext
+
+lookup_options = [href_option]
+
+repository_option = resource_option(
+    "--repository",
+    default_plugin="python",
+    default_type="python",
+    context_table=PulpRepositoryContext.TYPE_REGISTRY,
+    href_pattern=PulpRepositoryContext.HREF_PATTERN,
+    help=_("Repository to scan in the form '[<plugin>:<resource_type>:]<name>' or by href."),
+    required=True,
+)
+
+
+@pulp_group()
+@pass_pulp_context
+@click.pass_context
+def vulnerability_report(ctx: click.Context, pulp_ctx: PulpCLIContext, /) -> None:
+    ctx.obj = PulpVulnerabilityReportContext(pulp_ctx)
+
+
+vulnerability_report.add_command(list_command())
+vulnerability_report.add_command(show_command(decorators=lookup_options))
+
+
+@vulnerability_report.command()
+@repository_option
+@click.option(
+    "--version",
+    type=int,
+    help=_("Version of the repository to scan. Leave blank for latest version."),
+)
+@pass_pulp_context
+def create(
```

Comment:
I think, here we should follow the design of the api.
The repository version command group can have a `scan` subcommand whose sideeffect would be to create a report.